File: frontend/src/api/notification.ts
import { request } from './request'

export interface NotificationItem {
  id: number
  user_id: number
  sender_name: string
  sender_avatar?: string
  article_id: number
  article_title: string
  article_slug: string
  reply_content: string
  parent_content: string
  is_read: boolean
  created_at: string
}

export const getMyNotificationsApi = () => {
  return request<NotificationItem[]>({
    url: '/notifications/my',
    method: 'GET'
  })
}

export const getUnreadNotificationCountApi = () => {
  return request<number>({
    url: '/notifications/unread-count',
    method: 'GET'
  })
}

export const markAllNotificationsAsReadApi = () => {
  return request<null>({
    url: '/notifications/read-all',
    method: 'PUT'
  })
}

export const markNotificationAsReadApi = (id: number) => {
  return request<null>({
    url: `/notifications/read/${id}`,
    method: 'PUT'
  })
}

File: backend/app/schemas/notification.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class NotificationOut(BaseModel):
    id: int
    user_id: int
    sender_name: str
    sender_avatar: Optional[str] = None
    article_id: int
    article_title: str
    article_slug: str
    reply_content: str
    parent_content: str
    is_read: bool
    created_at: datetime

    class Config:
        from_attributes = True
